core/modules/replica/WorkerServer.cc

```diff
@@ -0,0 +1,129 @@
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/WorkerServer.h"
+
+// System headers
+
+// Third party headers
+#include <boost/bind.hpp>
+
+// Qserv headers
+#include "lsst/log/Log.h"
+#include "replica/Configuration.h"
+#include "replica/ServiceProvider.h"
+#include "replica/WorkerProcessor.h"
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.WorkerServer");
+
+} /// namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+WorkerServer::Ptr WorkerServer::create(ServiceProvider::Ptr const& serviceProvider,
+                                           WorkerRequestFactory& requestFactory,
+                                           std::string const& workerName) {
+    return WorkerServer::Ptr(
+        new WorkerServer(
+            serviceProvider,
+            requestFactory,
+            workerName));
+}
+
+WorkerServer::WorkerServer(ServiceProvider::Ptr const& serviceProvider,
+                           WorkerRequestFactory& requestFactory,
+                           std::string const& workerName)
+    :   _serviceProvider(serviceProvider),
+        _workerName(workerName),
+        _processor(serviceProvider, requestFactory, workerName),
+        _workerInfo(serviceProvider->config()->workerInfo(workerName)),
+        _io_service(),
+        _acceptor(
+            _io_service,
+            boost::asio::ip::tcp::endpoint(
+                boost::asio::ip::tcp::v4(),
+                _workerInfo.svcPort)) {
+
+    // Set the socket reuse option to allow recycling ports after catastrifc
+    // failures.
+
+    _acceptor.set_option(boost::asio::socket_base::reuse_address(true));
+}
+
+void WorkerServer::run() {
+
+    // Start the processor to allow processing requests.
+
+    _processor.run();
+
+    // Begin accepting connections before running the service to allow
+    // asynchronous operations. Otherwise the service will finish right
+    // away.
+
+    beginAccept();
+
+    _io_service.run();
+}
+
+void WorkerServer::beginAccept() {
+
+    WorkerServerConnection::Ptr connection =
+        WorkerServerConnection::create(
+            _serviceProvider,
+            _processor,
+            _io_service
+        );
+        
+    _acceptor.async_accept(
+        connection->socket(),
+        boost::bind (
+            &WorkerServer::handleAccept,
+            shared_from_this(),
+            connection,
+            boost::asio::placeholders::error
+        )
+    );
+}
+
+void WorkerServer::handleAccept(WorkerServerConnection::Ptr const& connection,
+                                boost::system::error_code const& ec) {
+
+    if (not ec) {
+        connection->beginProtocol();
+    } else {
+
+        // TODO: Consider throwing an exception instead. Another option
+        //       would be to log the message via the standard log file
+        //       mechanism since its' safe to ignore problems with
```

Comment:
it's

Reply:
Sometimes my fingers are disconnected from my brain :-) Fixed.